File: hw/Pérez/U2/HW17TestCaseSolution/ec.edu.espe.complexoperations.controller/PersonController.js
import { differenceInYears, differenceInMonths, differenceInDays } from 'date-fns';

class PersonController {
    static computeAgeInDays(birthDate) {
        const today = new Date();
        const birthDateObj = new Date(birthDate);

        const numberOfDaysOfYears = differenceInYears(today, birthDateObj) * 365;
        const numberOfDaysOfMonths = differenceInMonths(today, birthDateObj) % 12 * 30;
        const numberOfDays = differenceInDays(today, birthDateObj) % 30;

        const days = numberOfDaysOfYears + numberOfDaysOfMonths + numberOfDays;
        return days;
    }

    static computeAgeInYears(birthDate) {
        const today = new Date();
        const birthDateObj = new Date(birthDate);
        const numberOfDaysOfYears = differenceInYears(today, birthDateObj);
        return numberOfDaysOfYears;
    }
}

export default PersonController;

File: ws/travez/u2/ws23UnitTest/ComplexOperation/src/main/java/ec/edu/espe/complexoperation/controller/PersonController.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.complexoperation.controller;

import java.time.LocalDate;
import java.time.Period;

/**
 *
 * @author G406
 */
public class PersonController {

    public static int computeAgeInDays(LocalDate birthDate) {
        int days;
        LocalDate today = LocalDate.now();
        int numberOfDaysOfYears = Period.between(birthDate, today).getYears() * 365;
        int numberOfDaysOfMonths = (Period.between(birthDate, today).getMonths()) * 30;
        int numberOfDays = Period.between(birthDate, today).getDays() * 365;
        days = numberOfDaysOfYears + numberOfDaysOfMonths + numberOfDays;
        return days;
    }

    public static int computeAgeInYears(LocalDate birthDate) {
        LocalDate today = LocalDate.now();
        int numberOfDaysOfYears = Period.between(birthDate, today).getYears();
        return numberOfDaysOfYears;
    }
    
}
